### PR TITLE
DetailsList: address onItemInvoked bug by instructing FocusZone to avoid eating the enter keydown event

### LINKED ISSUE #14947: DetailsList onItemInvoked callback should be called when pressing Enter.
In documentation it is said that onItemInvoked is a "Callback for when a given row has been invoked (by pressing enter while it is selected.)" But in reality onItemInvoked is invoked only when the item is double clicked but not when Enter key is pressed. The callback should be invoked when Enter key is pressed (or consider making another callback that is invoked by Enter key).

### PR DESCRIPTION
**Pull request by dzearing**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #14947
- [X] Include a change request file using `$ yarn change`

#### Description of changes

A previous change to DetailsList re-ordered SelectionZone and FocusZone, which broke SelectionZone properly handling the enter keydown event. Partially reverting this change.

Added a unit test to cover so this doesn't bust again.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx [hunk 1]
```diff
@@ -109,6 +109,6 @@ export class DetailsListBasicExample extends React.Component<{}, IDetailsListBas
   };
 
   private _onItemInvoked = (item: IDetailsListBasicExampleItem): void => {
-    alert(`Item invoked: ${item.name}`);
+    console.log(`Item invoked: ${item.name}`);
```
Comment (JustSlone) [inline packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx:112]: do we need to update any docs that refer to this being an alert? i.e. is there text that says "double click to see an alert". I'm guessing no, but thought I would ask.

**Review by JustSlone — APPROVED** [review]
Looks good to me, thanks for adding tests!

File: packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx (hunk 1 above)
Comment (dzearing) [reply packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.Basic.Example.tsx]: done. :)

File: packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx [hunk 2]
```diff
@@ -505,6 +505,7 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
     direction: FocusZoneDirection.vertical,
     shouldEnterInnerZone: isRightArrow,
     onActiveElementChanged: onActiveRowChanged,
+    shouldRaiseClicks: false,
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx:508]: This is the 1 line fix that resolves this.

**Review by ThomasMichon — APPROVED** [review]

**Comment by brandonthomas** [comment]
Thanks a million for the quick turn around on this one guys!

**Pull request merged**